mlflow/pyfunc/__init__.py

```diff
@@ -1054,6 +1127,46 @@ def spark_udf(spark, model_uri, result_type="double", env_manager=_EnvManager.LO
 
     model_metadata = Model.load(os.path.join(local_model_path, MLMODEL_FILE_NAME))
 
+    model_output_schema = model_metadata.get_output_schema()
+
+    if result_type is None:
+        if model_output_schema is None:
+            _logger.warning(
+                "No 'result_type' provided for spark_udf and the model metadata does not "
+                "have an output schema. 'result_type' is set to 'double' type."
+            )
+            result_type = "double"
+        else:
+            result_type = _infer_spark_udf_return_type(model_output_schema)
+    else:
+        result_type = "boolean" if result_type == "bool" else result_type
```

Comment:
Nit: do we need the `else` here?

```suggestion

    result_type = "boolean" if result_type == "bool" else result_type
```

should be fine, right?